### DISCUSSION TITLE
1.1.0 Release candidate feedback

### DISCUSSION THREAD

**Discussion by tmijail** [tmijail/Odoo-Power-BI-Connector#24] [General]
### Call for testers
This is an experimental release. If you have reports running on v1.0.0, please try this release and let me know in the associated discussion if you experience any issues or if everything works well.

If you are a new user, use the [latest stable release](https://github.com/tmijail/Odoo-Power-BI-Connector/releases/latest) instead.

### Changes
 - Added query folding support for `Table.FirstN`
 - Added query folding support for `Table.Skip`
 - Added query folding support for `Table.SelectColumns`
 - Added query folding support for `Table.Sort`

### Explanation
As the [Power Query documentation](https://learn.microsoft.com/en-us/power-query/query-folding-basics#query-folding-overview) says,

> The goal of query folding is to offload or push as much of the evaluation of a query to a data source that can compute the transformations of your query.
> 
> The query folding mechanism accomplishes this goal by translating your M script to a language that can be interpreted and executed by your data source. It then pushes the evaluation to your data source and sends the result of that evaluation to Power Query.
> 
> This operation often provides a much faster query execution than extracting all the required data from your data source and running all transforms required in the Power Query engine.

<hr /><em>This discussion was created from the release <a href='https://github.com/tmijail/Odoo-Power-BI-Connector/releases/tag/'>1.1.0 Release candidate</a>.</em>

**Comment by amoradel** [comment]
Hola @tmijail 

Para una consulta como esta:
`= search_read("account.move.line",{{"parent_state","=", "posted"}, {"date",">=", "08/01/2023"}, {"date","<", "10/01/2023"}}, [context=[lang="es_GT"], fields = {"account_id", "debit", "credit", "date", "full_reconcile_id", "analytic_account_id"}])`

¿Se pueden aplicar los cambios de query folding que desarrollaste?

**Reply by tmijail** [reply]
Buenos días, Anibal.

Si partes de esa consulta y en pasos siguientes usas la UI de PowerBI para agregar o quitar columnas, ordenar, limitar o saltear resultados se aplica el query folding manteniendo los filtros y contexto que especificaste manualmente. No hay ningún problema con mezclar consultas escritas a mano con operaciones por UI.

Lo que no se puede hacer actualmente es aplicar filtros (`{"parent_state","=", "posted"}`, etc.) mediante query folding. Es teóricamente posible, pero no está implementado.

**Reply by amoradel** [reply]
Entendido Mijail,

Muchas gracias.

**Comment by Rog15227** [comment]
@tmijail Hi,

First of all many thanks for your work and the connector you've developped that helped us a lot.

Unfortunately, since a last update that appears the 16th October we're facing a new issue that we've provided to our Odoo consultant. Here's their response:

_Hello,

Thank you for contacting Odoo support and for your patience so far, as we've been facing a recent influx of tickets. I understand that you were utilizing PowerBI to analyze data from your database and found that a field is not returning a value (that did previously).

In this case, the code of Odoo modules indeed change between updates and upgrades. From my review, the field "l10n_din5008_addresses" is related to the addresses used with the German (l10n_de) localization package. I am unsure what exactly the field is used for, but I wanted to link the relevant code that creates the field (https://github.com/odoo/odoo/blob/16.0/addons/l10n_din5008_sale/models/sale.py). It seems this field is used for reports related to the German Localization and it seems to capture the various address of a Contact on a Sales Order (for example) to be displayed on the German Localization Reports.

Regarding the behavior seen (not being able to export the field), I think this is related to this commit (https://github.com/odoo/odoo/commit/72e9f302389febd1528bab2573882eff321ddb08). It seems there was a change made to the "l10n_din5008_addresses" making it not exportable by default. This is likely causing the behavior seen.

Unfortunately in this case, reviewing accessing a database via API and transfer of data to an external source (such as PowerBI) is outside of my support scope. I would need to refer you to work with an Odoo consultant to review your connection to an Odoo database.

The code of modules can be changed/updated which can affect external connections. Fields may be renamed, their properties changed, or even completely removed. The most accurate way to determine what has changed would be to inspect the GitHub repository where the field is created and inspect for commits changing the behavior. I understand this may be technical challenging, but I wanted to share this information so you can utilize this in case you encounter a similar situation and have a reference point.

I hope this makes sense and is helpful for you! Please let me know what other questions you may have regarding this. Thank you!_

As you can see, is there any solutions that can help us to rearrange our requests regarding those kind of situations in Power BI when they change the data structure ?
We've tried so far some communatory app which wasn't very successfull to be honest.

Sorry in advance for my request, which may seem absurd. I'm trying to develop automated reports in the company where I work, and I'm doing it entirely self-taught without prior knowledge in databases; it's quite new to me.

**Comment by syedmaz22** [comment]
Primero que nada muchas gracias.

En segundo lugar, recibo este error cuando consulto solo el modelo "pos.order". Otras tablas funcionan bien

Error del servidor Odoo: el objeto '_unknown' no tiene el atributo 'id'
